### PR TITLE
Add missing 2021 Utah parameter values

### LINKED ISSUE #2180: Add 2021 Utah retirement credit and taxpayer credit parameters
The Utah state income tax code has several problems.  The integration test presented below illustrates two of those problems.   

**First**, the taxpayer credit is incorrect because PolicyEngineUS projects the 2022 phase-out threshold value (which is correct) back to 2021.   This is happening even though earlier in the development of the Utah income tax the 2021 parameter values were in the code.  This can be seen in the following screenshot of a March 23rd code review session for pull request #1994.

Screenshot 2023-04-25 at 9 48 15 AM

When #1994 was merged a month later, the 2021 values had been removed.  This is just one example of the many missing 2021 parameter values.  The 2021 parameter values are **missing** because some Utah tax parameters are specified as far back as 2020, 2018, and even 2008.

**Second**, the retirement credit is incorrect.  PolicyEngineUS generates a retirement credit for this test case that is zero when following the TC-40 form instructions produces a value of $624.50.  The $624.50 retirement credit is also generated by TAXSIM35.

Here is the integration test case followed by the test results:
```
- name: Tax unit with taxsimid 95281 in p21.its.csv and p21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 73
        employment_income: 21_010
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        is_tax_unit_spouse: true
        age: 73
        employment_income: 22_010
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2]
        premium_tax_credit: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2]
        state_code: UT
  output:  # expected results from patched TAXSIM35 2023-04-24 version
    ut_income_tax: 3.78
```
This test case produces these results:
```
  Test 'Tax unit with taxsimid 95281 in p21.its.csv and p21.ots.csv':
    ut_income_tax@2021: [616.50195] differs from 3.7799999713897705 
                        with an absolute margin [612.7219] > 0.01
```

### PR DESCRIPTION
**Pull request by XZ-mabel**
### Summary
📝🚨🏛️

This pull request updates the Utah retirement credit parameter `max` in `policyengine_us/parameters/gov/states/ut/tax/income/credits/retirement/max.yaml` to reflect the current law. It also adds a reference to the source of the parameter value.

> _To match the current law of the state_
> _We updated the `max` parameter rate_
> _For the Utah retirement credit_
> _We added a reference to edit_
> _The statute that we can now cite and relate_

### Walkthrough
* Update the effective date of the Utah retirement credit to 2021, to match the actual legislation ([link](https://github.com/PolicyEngine/policyengine-us/pull/3061/files?diff=unified&w=0#diff-5c868bb48add5ca4a7cf276bf4e2c4fbe7d4c2af7bcecd3dad881038d7eb8ebaL3-R3))
* Add a reference to the Utah Code section that defines the retirement credit, to provide a source for the parameter value and the eligibility criteria ([link](https://github.com/PolicyEngine/policyengine-us/pull/3061/files?diff=unified&w=0#diff-5c868bb48add5ca4a7cf276bf4e2c4fbe7d4c2af7bcecd3dad881038d7eb8ebaR9-R10))

Fixes #2180

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml [hunk 1]
```diff
@@ -3,4 +3,10 @@ values:
   2022-01-01: 100
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml:3]: need change

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml:1]: make a non_refundable.yaml file with all the non refundable credits from ut_income_tax_before_refundable_credits

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml:1]: remove the if statement from the credit computation

**Comment by MaxGhenis** [comment]
tests are failing

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need git pull upstream master

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Utah matches this percent of the federal EITC.
+values:
+  2022-01-01: 0.15
+metadata:
+  unit: /1
+  label: Utah EITC match
+  reference:
+    - title: Utah legal code| UC §59-10-1044
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:8]:
```suggestion
    - title: Utah legal code| UC §59-10-1044 (2) (a)
```

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+description: Utah matches this percent of the federal EITC.
+values:
+  2022-01-01: 0.15
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:3]: isn't it 20% ?

image

Comment (XZ-mabel) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:3]: I think the 15% comes from here: https://incometax.utah.gov/credits/earned-income-tax-credit#:~:text=You%20may%20claim%20a%20Utah,in%20the%20same%20tax%20year. Should we use 20% here?

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (XZ-mabel) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:8]: I think the credit was introduced in 2022, based on this: https://incometax.utah.gov/media/eitc-slides.pdf#page=4. This pdf is on this website: https://incometax.utah.gov/credits/earned-income-tax-credit#:~:text=You%20may%20claim%20a%20Utah,in%20the%20same%20tax%20year.

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 description: The amount of at-home credit per qualifying child.
 values:
-  2022-01-01: 100
+  2021-01-01: 100
 metadata:
   unit: currency-USD
+  period: year
   label: Utah at-home parent credit amount
+  reference:
+    - title: Utah Code §59-10-1005
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml:9]: add section of the code

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/amount.yaml:9]: for all yaml files

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:3]: what does the tax form say? Maybe it is 20% in 2023

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml]: could you add a 2021 tax form and add a comment " # the credit was introduced in 2022"

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 3 above)
Comment (XZ-mabel) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:3]: It says 15% in the instructions for the 2022 tax form: https://tax.utah.gov/forms/current/tc-40inst.pdf#page=22

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml]: You are absolutely right, can you add this source as well showing the 15% in 2022 https://law.justia.com/codes/utah/2022/title-59/chapter-10/part-10/section-1044/

Also add a 2023-01-01: 0.2 and a comment over the initial legal source saying that it increased in 2023 to 20%

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:3]: You are absolutely right, can you add this source as well showing the 15% in 2022 https://law.justia.com/codes/utah/2022/title-59/chapter-10/part-10/section-1044/

Also add a 2023-01-01: 0.2 and a comment over the initial legal source saying that it increased in 2023 to 20%

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  2023-01-01: 0.2
+metadata:
+  unit: /1
+  label: Utah EITC match
+  reference:
+    - title: Utah legal code| UC §59-10-1044 (2) (a)  #The percentage ncreased in 2023 to 20%
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:9]:
```suggestion
    - title: Utah legal code| UC §59-10-1044 (2) (a)  # The percentage increased in 2023 to 20%
```

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      href: https://le.utah.gov/xcode/Title59/Chapter10/59-10-S1044.html 
+    - title: Utah TC-40  Utah Individual Income Tax Form #The credit was introduced in 2022
+      href: https://www.taxformfinder.org/forms/2021/2021-utah-tc-40-full-packet.pdf 
+    - title: 2022 Utah legal code| 59-10-1044
+      href: https://law.justia.com/codes/utah/2022/title-59/chapter-10/part-10/section-1044/
+    
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:16]: delete extra line

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  2023-01-01: 0.2
+metadata:
+  unit: /1
+  label: Utah EITC match
+  reference:
+    - title: Utah legal code| UC §59-10-1044 (2) (a)  #The percentage ncreased in 2023 to 20%
+      href: https://le.utah.gov/xcode/Title59/Chapter10/59-10-S1044.html 
+    - title: Utah TC-40  Utah Individual Income Tax Form #The credit was introduced in 2022
+      href: https://www.taxformfinder.org/forms/2021/2021-utah-tc-40-full-packet.pdf 
+    - title: 2022 Utah legal code| 59-10-1044
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/nonrefundable/eitc.yaml:13]: add section

File: policyengine_us/parameters/gov/states/ut/tax/income/rate.yaml [hunk 8]
```diff
@@ -3,18 +3,21 @@ values:
...
         href: https://le.utah.gov/xcode/historical.html?date=5/8/2018&oc=/xcode/Title59/Chapter10/C59-10-S104_2018050820180508.html
   2022-01-01:
     value: 0.0485
     reference:
-      - title: Utah Code 59-10-104
+      - title: Utah Code 59-10-104 (2) (b)
         href: https://le.utah.gov/xcode/Title59/Chapter10/59-10-S104.html?v=C59-10-S104_2022050420220504
+      - title: 2023 Utah Code 59-10-104 (2) (b) # The rate was changed to 0.0465 in May 2023
+        href: https://le.utah.gov/xcode/Title59/Chapter10/59-10-S104.html?v=C59-10-S104_2022050420220504
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ut/tax/income/rate.yaml:20]: can you restructure this parameter to be:
  2008-01-01: 
    value: 0.05
  2018-05-08:
    value: 0.0495
  2022-01-01:
    value: 0.0485
  2022-01-01:
    value: 0.0465 
    
    and then the references on the bottom

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ut/tax/income/rate.yaml [hunk 9]
```diff
@@ -2,19 +2,21 @@ description: Utah main income tax rate.
...
   2022-01-01:
     value: 0.0485
-    reference:
-      - title: Utah Code 59-10-104
-        href: https://le.utah.gov/xcode/Title59/Chapter10/59-10-S104.html?v=C59-10-S104_2022050420220504
+  2023-01-01:
+    value: 0.0465
 metadata:
-  unit: /1
+  unit: bool
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ut/tax/income/rate.yaml:12]: ?

File: policyengine_us/variables/gov/states/ut/tax/income/ut_non_refundable.py [hunk 10]
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/ut_non_refundable.py:1]: Rename ut_non_refundable_credits

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit.py [hunk 11]
```diff
@@ -7,22 +7,6 @@ class ut_retirement_credit(Variable):
...
-            - tax_unit("ut_eitc", period)
-        )
-        if not parameters(
-            period
-        ).gov.states.ut.tax.income.credits.retirement.refundable:
-            return min_(
-                claims_retirement_credit * max_credit, limiting_liability
-            )
-        return claims_retirement_credit * max_credit
+    adds = ["ut_retirement_credit_max"]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit.py:12]: @nikhilwoodruff you did utah previous right? could you remind me why we did the non-refundability this way initially?

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit.py:12]: 👋

Comment (nikhilwoodruff) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit.py:12]: This was the general pattern for all nonrefundable credits at the time right?

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**